src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -127,12 +135,31 @@ public function getEnv($prefix, $name, \Closure $getEnv)
         } elseif (isset($_SERVER[$name]) && 0 !== strpos($name, 'HTTP_')) {
             $env = $_SERVER[$name];
         } elseif (false === ($env = getenv($name)) || null === $env) { // null is a possible value because of thread safety issues
-            if (!$this->container->hasParameter("env($name)")) {
-                throw new EnvNotFoundException(sprintf('Environment variable not found: "%s".', $name));
+            foreach ($this->loadedVars as $vars) {
+                if (false !== $env = ($vars[$name] ?? false)) {
+                    break;
+                }
             }
 
-            if (null === $env = $this->container->getParameter("env($name)")) {
-                return null;
+            try {
+                while ((false === $env || null === $env) && $this->loaders->valid()) {
```

Comment:
actually, it does trigger an exception `Cannot resume an already running generator`
reproduct case:
- a service need a secret `%enfFOO)%`
- this line iterate over registered env_var_loader (with DIC service locator)
- by default, the SodiumVault requires an env variable defined by config option `decryption_env_var`
- if the env variable does not exists, this is line is reached a second time and trigger the exception.

Reply:
fixed in https://github.com/symfony/symfony/pull/34301